lib/ask/runtime/step.ex

```diff
@@ -165,18 +165,12 @@ defmodule Ask.Runtime.Step do
       "numeric" ->
         # Only send numDigits if the min and max values have the same string length,
         # also taking into account the values of refusal responses
-        refusal = step["refusal"]
-
-        values = if refusal do
-          refusal["responses"]["ivr"]
-        else
-          []
-        end
-
         min_value = step["min_value"]
         max_value = step["max_value"]
+        refusal_values = fetch(:refusal, step, "ivr", language) || []
+
         if min_value && max_value do
-          values = [min_value, max_value | values]
+          values = [min_value, max_value | refusal_values]
           |> Enum.map(fn v -> v |> to_string |> String.length end)
           |> Enum.uniq
 
```

Comment:
We can improve code in lines 177-181:
```
  if length(values) == 1 do
    hd(values)
  else
    nil
  end
```
with: 
```
List.first(values)
```
Which returns `nil` on `empty` lists
https://hexdocs.pm/elixir/List.html#first/2

Reply:
We can't use `List.first/2` because there must be _exactly_ one remaining entry in the list.